js/engine.js

```diff
@@ -705,7 +758,35 @@ class Engine {
     const convertedDelta = unitConverter.convert(amount, currentValue.getUnits());
     const newAmount = currentValue.getValue() + convertedDelta.getValue();
     const outputWithUnits = new EngineNumber(newAmount, currentValue.getUnits());
-    self.setStream(stream, outputWithUnits, null, scope);
+    // Pass the original user-specified units for tracking
+    self.setStream(stream, outputWithUnits, null, scope, true, amount.getUnits());
+  }
+
+  /**
+   * Internal method to change a stream value without tracking units.
+   * Used by cap, floor, replace methods that handle units tracking themselves.
+   *
+   * @param {string} stream - The stream identifier to modify.
+   * @param {EngineNumber} amount - The amount to change the stream by.
+   * @param {Object} yearMatcher - Matcher to determine if the change applies
+   *     to current year.
+   * @param {Scope} scope - The scope in which to make the change.
+   */
+  _changeStreamInternal(stream, amount, yearMatcher, scope) {
```

Comment:
What do you think of changing this to `_changeStreamWithoutReportingUnits` to describe why one might use it?